File: backend/src/main/java/br/ufpr/sept/so2/modules/academico/api/dto/CursoRequest.java
package br.ufpr.sept.so2.modules.academico.api.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import java.util.UUID;

public record CursoRequest(
        @NotBlank @Size(max = 200) String nome,
        @NotBlank @Size(max = 20) String sigla,
        @NotBlank @Size(max = 30) String codigo,
        UUID idCoordenador,
        @Min(0) Integer horasFormativasMinimas,
        Boolean ativo
) {
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/academico/api/dto/CursoRequest.kt
package br.ufpr.sept.so2.modules.academico.api.dto

import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size
import java.util.UUID

data class CursoRequest(
    @field:NotBlank @field:Size(max = 200) val nome: String? = null,
    @field:NotBlank @field:Size(max = 20) val sigla: String? = null,
    @field:NotBlank @field:Size(max = 30) val codigo: String? = null,
    val idCoordenador: UUID? = null,
    @field:Min(0) val horasFormativasMinimas: Int? = null,
    val ativo: Boolean? = null,
)
